ui/components/multichain/connected-status/connected-status.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import {
+  BackgroundColor,
+  BorderColor,
+  Color,
+} from '../../../helpers/constants/design-system';
+import {
+  getAddressConnectedSubjectMap,
+  getOriginOfCurrentTab,
+} from '../../../selectors';
+import {
+  STATUS_CONNECTED,
+  STATUS_CONNECTED_TO_ANOTHER_ACCOUNT,
+  STATUS_NOT_CONNECTED,
+} from '../../../helpers/constants/connected-sites';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { BadgeStatus } from '../badge-status';
+
+export const ConnectedStatus = ({ address = '', isActive = false }) => {
+  const t = useSelector(useI18nContext);
+
+  const addressConnectedSubjectMap: any = useSelector(
+    getAddressConnectedSubjectMap,
+  );
+  const originOfCurrentTab = useSelector(getOriginOfCurrentTab);
+
+  const selectedAddressSubjectMap = addressConnectedSubjectMap[address];
+  const currentTabIsConnectedToSelectedAddress = Boolean(
+    selectedAddressSubjectMap?.[originOfCurrentTab],
+  );
+
+  let status;
+  if (isActive) {
+    status = STATUS_CONNECTED;
+  } else if (currentTabIsConnectedToSelectedAddress) {
+    status = STATUS_CONNECTED_TO_ANOTHER_ACCOUNT;
+  } else {
+    status = STATUS_NOT_CONNECTED;
```

Comment:
Instead of this `else` we can just initialize the value:

```
let status = STATUS_NOT_CONNECTED;
```

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/23006/commits/58f909010573d635f64dc02d58508fb316309d97